### DISCUSSION TITLE
Update nowplaying metadata via API

### DISCUSSION THREAD

**Discussion by fluential** [AzuraCast/AzuraCast#5276] [Q&A]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Stable Channel

### Current AzuraCast Version

v0.15.2 Stable • Docker • PHP 8.1

### What happened?

On a fresh vanilla installation, I am trying to follow usual way of updating nowplaying metadata via API like it is described here 
https://github.com/AzuraCast/AzuraCast/issues/2801#issuecomment-626309820

```
curl -v http://your-azuracast-install/api/station/(station_id)/nowplaying/update?title=YourTitle&api-key=$KEY"
Error: Permission denied. on /var/azuracast/www/src/Middleware/Permissions.php L40
```
the update fails with `Permission denied. on /var/azuracast/www/src/Middleware/Permissions.php L40` I am using SuperAdmin API key. The same problem exist when I create dedicated station user with all the required "station broadcast" permissions.

I've discovered this when testing before update from some older version (v0.12.4 Stable) where this works fine. 
Thoughts?

### Relevant log output

```shell
curl -v http://your-azuracast-install/api/station/(station_id)/nowplaying/update?title=YourTitle&api-key=$KEY"
Error: Permission denied. on /var/azuracast/www/src/Middleware/Permissions.php L40
```
```

**Comment by Vaalyn** [comment]
You'll need to supply the API Key via a header for it to work.
You can either use the `X-API-Key` or `Authorization` header.

Examples:
- `curl ... -H 'X-API-Key: xxxxx:xxxx'`
- `curl ... -H 'Authorization: Bearer xxxxx:xxxx'`

Here an curl example a user posted on our Discord (using POST instead of GET):
```
curl -X 'POST' 'https://radio.example.com/api/station/1/nowplaying/update' -H 'Content-Type: application/json' -H 'X-API-Key: xxxxx:xxxxxxx' -d '{"title": "My Song", "artist": "My Artist"}'
```

I just tested both `GET` and `POST` requests with both of these headers works correctly.
